File: src/01-easy/retro-typewriter-art/solution.kt
fun main(){
    val ab=mapOf("sp" to ' ',"bS" to '\\',"sQ" to '\'',"nl" to '\n')
    val line=readLine()!!
    val sb=StringBuilder()
    for(tok in line.split(" ")){
        val l2=if(tok.length>=2) tok.takeLast(2) else ""
        val (ch,num)=if(ab.containsKey(l2)) Pair(ab[l2]!!,tok.dropLast(2)) else Pair(tok.last(),tok.dropLast(1))
        val n=if(num.isEmpty()) 1 else num.toInt().coerceAtLeast(1)
        repeat(n){sb.append(ch)}
    }
    println(sb)
}

File: src/01-easy/retro-typewriter-art/solution.swift
import Foundation
let ab: [String: Character] = ["sp": " ", "bS": "\\", "sQ": "'", "nl": "\n"]
let line = readLine()!
var out = ""
for tok in line.split(separator: " ") {
    let s = String(tok)
    let l2 = s.count >= 2 ? String(s.suffix(2)) : ""
    let ch: Character; let num: String
    if let c = ab[l2] { ch = c; num = String(s.dropLast(2)) }
    else { ch = s.last!; num = String(s.dropLast(1)) }
    let n = max(1, num.isEmpty ? 1 : Int(num)!)
    out += String(repeating: ch, count: n)
}
print(out)
